StackExchange.Redis/StackExchange/Redis/GeoEntry.cs

```diff
@@ -0,0 +1,284 @@
+        /// </summary>
+        public double Radius => radius;
+        /// <summary>
+        /// The center point to base the search.
+        /// </summary>
+        public GeoPosition GeoPosition => geoPosition;
+        /// <summary>
+        /// The key to use.
+        /// </summary>
+        public RedisKey Key => key;
+        /// <summary>
+        /// The unit to return distance measurments in.
+        /// </summary>
+        public GeoUnit GeoUnit => geoUnit;
+        /// <summary>
+        /// The possible options for the GeoRadius command
+        /// </summary>
+        public GeoRadiusOptions GeoRadiusOptions => geoRadiusOptions;
+        /// <summary>
+        /// The maximum number of results to return.
+        /// However note that internally the command needs to perform an effort proportional to the number of items matching the specified area, so to query very large areas with a very small COUNT option may be slow even if just a few results are returned.
+        /// </summary>
+        public int MaxReturnCount => maxReturnCount;
+        /// <summary>
+        /// Creates a new GeoRadius
+        /// </summary>
+        public GeoRadius(RedisKey key,GeoPosition geoPosition,double radius,int maxReturnCount =-1,GeoUnit geoUnit = GeoUnit.Meters,GeoRadiusOptions geoRadiusOptions = (GeoRadiusOptions.WITHCOORD | GeoRadiusOptions.WITHDIST | GeoRadiusOptions.WITHHASH))
+        {
+            this.key = key;
+            this.geoPosition = geoPosition;
+            this.radius = radius;
+            this.geoUnit = geoUnit;
+            this.geoRadiusOptions = geoRadiusOptions;
+            this.maxReturnCount = maxReturnCount;
+        }
+
+    }
+
+    /// <summary>
+    /// Describes the longitude and latitude of a GeoEntry
+    /// </summary>
+    public struct GeoPosition
+    {
+        internal readonly double longitude;
+        internal readonly double latitude;
+
+        /// <summary>
+        /// The Latitude of the GeoPosition
+        /// </summary>
+        public double Latitude => latitude;
+        /// <summary>
+        /// The Logitude of the GeoPosition
+        /// </summary>
+        public double Longitude => longitude;
+
+        /// <summary>
+        /// Creates a new GeoPosition
+        /// </summary>
+        /// <param name="longitude"></param>
+        /// <param name="latitude"></param>
+        public GeoPosition(double longitude,double latitude)
+        {
+            this.longitude = longitude;
+            this.latitude = latitude;
+        }
+
+        /// <summary>
+        /// See Object.ToString()
+        /// </summary>
+        public override string ToString()
+        {
+            return string.Format("{0} {1}", longitude, latitude);
+        }
+        /// <summary>
+        /// See Object.GetHashCode()
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return longitude.GetHashCode() ^ latitude.GetHashCode();
```

Comment:
minor, but `^` tends to have problems with diagonals; would prefer a different implementation, but OTOH, I guess diagonals aren't going to come up *that* often...